packages/plugin-connector/src/data-sources/exec-source.js

```diff
@@ -0,0 +1,31 @@
+import { buildParquetFromResultSet } from '@evidence-dev/universal-sql';
+import fs from 'fs/promises';
+/**
+ *
+ * @param {DatasourceSpec} source
+ * @param {PluginDatabases} supportedDbs
+ */
+export const execSource = async (source, supportedDbs) => {
+	if (!(source.type in supportedDbs)) {
+		// TODO: Make this error message better
+		throw new Error(`Unsupported database type: ${source.type}`);
+	}
+
+	const db = supportedDbs[source.type];
+	const runner = await db.factory(source.options, source.sourceDirectory);
+	const results = await Promise.all(
+		source.queries.map(async (q) => {
+			return {
+				...q,
+				result: await runner(q.content, q.filepath)
+			};
+		})
+	);
+
+	for (const query of results) {
+		const { result } = query;
+		if (!result) continue;
+		const parquetBuffer = await buildParquetFromResultSet(result.columnTypes, result.rows);
+		await fs.writeFile(query.filepath.split('.')[0] + '.parquet', parquetBuffer);
```

Comment:
Any chance we could get file named `myFile.prod.sql` ?

Reply:
We could add the `.prod`, but we wouldn't want to use `.sql` here I would imagine; these are binary columnar files